lib/ex_sdp/origin.ex

```diff
@@ -8,27 +8,51 @@ defmodule ExSDP.Origin do
 
   For more details please see [RFC4566 Section 5.2](https://tools.ietf.org/html/rfc4566#section-5.2)
   """
+  use Bunch.Access
 
   alias ExSDP.ConnectionData
 
   @enforce_keys [
-    :username,
     :session_id,
     :session_version,
     :address
   ]
-  defstruct @enforce_keys
+  defstruct [username: "-"] ++ @enforce_keys
 
   @type t :: %__MODULE__{
           username: binary(),
           session_id: binary(),
           session_version: binary(),
-          # FIXME this cannot be ConnectionData.sdp_address() as it has no ttl field for IPv4
+          # FIXME this cannot be ConnectionData.sdp_address() as it can't have ttl field
           address: ConnectionData.sdp_address()
         }
 
   @type reason :: :invalid_address | ConnectionData.reason()
 
+  @doc """
+  Return new `%__MODULE{}` struct. `session_id` and `session_version` are strings representing
+  random 64 bit numbers.
+  """
+  @spec new_random(
+          username :: binary(),
+          session_id :: binary(),
+          session_version :: binary(),
+          address :: ConnectionData.sdp_address()
+        ) :: t()
+  def new_random(
+        username \\ "-",
+        session_id \\ generate_random(),
+        session_version \\ generate_random(),
+        address
```

Comment:
either move the optional arguments to the end or take them from a keyword list argument